Description: Ignore broadcasted PADO-packets
 Ignore broadcasted PADO-packets instead of packets from non-unique source
 address.
Author: Stephane Louise <luigi@nerim.net>
Bug-Debian: https://bugs.debian.org/293811
Forwarded: no
Reviewed-By: Andreas Barth <aba@not.so.argh.org>
Last-Update: 2016-06-20

Index: rp-pppoe/src/discovery.c
===================================================================
--- rp-pppoe.orig/src/discovery.c
+++ rp-pppoe/src/discovery.c
@@ -472,8 +472,8 @@ waitForPADO(PPPoEConnection *conn, int t
 	if (!packetIsForMe(conn, &packet)) continue;
 
 	if (packet.code == CODE_PADO) {
-	    if (NOT_UNICAST(packet.ethHdr.h_source)) {
-		printErr("Ignoring PADO packet from non-unicast MAC address");
+	    if (BROADCAST(packet.ethHdr.h_source)) {
+		printErr("Ignoring broadcast PADO packet");
 		continue;
 	    }
 #ifdef PLUGIN
